atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/queue/TargetedSweeper.java

```diff
@@ -241,6 +306,41 @@ private int getPartitionBatchLimit() {
                 : runtime.get().maximumPartitionsToBatchInSingleRead();
     }
 
+    @VisibleForTesting
+    void updateLastSweptTsMetric(SweeperStrategy sweeperStrategy) {
+        int shards = queue.getNumShards();
+
+        Set<ShardAndStrategy> shardAndStrategySet = IntStream.range(0, shards)
+                .mapToObj(shard -> ShardAndStrategy.of(shard, sweeperStrategy))
+                .collect(Collectors.toSet());
+
+        Map<ShardAndStrategy, Long> shardAndStrategyToTimestamp = queue.getLastSweptTimestamps(shardAndStrategySet);
+
+        KeyedStream.stream(shardAndStrategyToTimestamp).forEach(metrics::updateProgressForShard);
+    }
+
+    private class LastSweptTsUpdateTask implements AutoCloseable {
```

Comment:
You likely won't need this, but nit: final!

Separately, why is this AutoCloseable and the scheduler Closeable? Just curious

Reply:
No reason, I was just mirroring a similar use case in the code. Upon looking up the difference between the two, both classes should be AutoCloseable